build/dsc/templates/dsc_type.rb.erb

```diff
@@ -86,11 +86,15 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
 
   ensurable do
     newvalue(:exists?) { provider.exists? }
-    newvalue(:<%= begin resource.ensure_value rescue 'present' end %>) { provider.create }
+<%  resource.valid_ensure_present_values.each do |property| -%>
+    newvalue(:<%= property.downcase %>) { provider.create }
+<%  end -%>
 <%  if resource.absentable? -%>
-    newvalue(:<%= resource.absent_value %>)  { provider.destroy }
+<%  resource.valid_ensure_absent_values.each do |property| -%>
+    newvalue(:<%= property.downcase %>)  { provider.destroy }
```

Comment:
Implementing it kept a certain symmetry and was easy to do, so I followed the path. If there is variance with present, it stands to reason there could variance with absent. Argument goes both, ways: future pain or present effort. Don't have a preference.